discord-bot/src/commands/market-react-bets.ts

```diff
@@ -0,0 +1,147 @@
+import { FullMarket } from 'common/api-market-types'
+import {
+  AttachmentBuilder,
+  ChatInputCommandInteraction,
+  EmbedBuilder,
+  MessageReaction,
+  SlashCommandBuilder,
+  TextChannel,
+  User,
+} from 'discord.js'
+import { customEmojis, emojis, getBettingEmojisAsStrings } from '../emojis.js'
+import {
+  getCurrentMarketDescription,
+  getOpenBinaryMarketFromSlug,
+  getSlug,
+  handleReaction,
+} from '../helpers.js'
+import { messagesHandledViaInteraction } from '../storage.js'
+
+export const data = new SlashCommandBuilder()
+  .setName('market')
+  .setDescription('Link to a market that other users can bet on with reactions')
+  .addStringOption((option) =>
+    option
+      .setName('link')
+      .setDescription('The link to the market to bet on')
+      .setRequired(true)
+  )
+
+export async function execute(interaction: ChatInputCommandInteraction) {
+  const link = interaction.options.getString('link')
+  if (!link || !link.startsWith('https://manifold.markets/')) {
+    await interaction.reply(
+      'You must specify a market link starting with https://manifold.markets/'
+    )
+    return
+  }
+  const slug = getSlug(link)
+  if (!slug) {
+    await interaction.reply(
+      'Invalid market link, could not find slug from link'
+    )
+    return
+  }
+  const market = await getOpenBinaryMarketFromSlug(slug, (error) =>
+    interaction.reply(error)
+  )
+  if (!market) return
+
+  const message = await sendMarketIntro(interaction, market)
+
+  const filter = (reaction: MessageReaction, user: User) => {
+    if (user.id === message.author.id) return false
+    return !!reaction.emoji
+  }
+
+  const collector = message.createReactionCollector({ filter, dispose: true })
+  const channel = interaction.channel as TextChannel
+  collector.on('collect', async (reaction, user) => {
+    await handleReaction(reaction, user, channel, market)
+  })
+
+  // Removed the un react action for now
+  // collector.on('remove', async (reaction, user) => {
+  //   const message = reaction.message.partial
+  //     ? await reaction.message.fetch()
+  //     : reaction.message
+  //
+  //   await handleBet(reaction, user, channel, message, market, true)
+  // })
+}
+
+const sendMarketIntro = async (
+  interaction: ChatInputCommandInteraction,
+  market: FullMarket
+) => {
+  const placeHolderEmbed = new EmbedBuilder()
+    .setColor(0x0099ff)
+    .setTitle('Loading market...')
+  let message = await interaction.reply({
+    embeds: [placeHolderEmbed],
+    fetchReply: true,
+  })
+
+  // Let client listener know we've this message in memory
+  messagesHandledViaInteraction.add(message.id)
+
+  // Add emoji reactions
+  for (const emoji of emojis) {
+    if (customEmojis.includes(emoji)) {
+      // TODO: this only works on my guild rn
+      const reactionEmoji = interaction.guild?.emojis.cache.find(
+        (e) => e.id === emoji
+      )
```

Comment:
I wonder if this cache is populated right away on startup?

Reply:
Not sure! Either way we have to figure out how to use the same emojis across different guilds